### PR TITLE
Crew Monitor Robotic Races Update

### PR DESCRIPTION
**Pull request by Deek-Za**
## About The Pull Request

Modifies crew.dm and CrewConsole.js to update the crew monitor with a flag for any robotic races (IPC, Synthliz, Synthhuman, etc.)

## How This Contributes To The Skyrat Roleplay Experience

Well here on Skyrat we have a ton of synth races and players who play them.
Ever see someone injured on the crew monitor, grab your medkit and some chems and rush out to help save the day then.. Uh oh. He's a synth. So all that stuff you brought is **useless**. I know I have a time or two again. And we got new friendly robot faces popping up everyday!

This just adds a little cog icon if they aren't organic.
image

Icon updated to:
image

## Changelog

:cl: Deek-Za
qol: Flags non-organic races on the crew monitor.
/:cl:

### CODE REVIEW

**Comment by Zonespace27** [comment]
Since it's a nonmodular TGUI change, please take it upstream first.

**Comment by Elli-Skala** [comment]
> Since it's a nonmodular TGUI change, please take it upstream first.

Tg doesn't appear to have any non adminbussed in synth race. So this would be quite useless to them

**Comment by Zonespace27** [comment]
> > Since it's a nonmodular TGUI change, please take it upstream first.
> 
> Tg doesn't appear to have any non adminbussed in synth race. So this would be quite useless to them

TG has IPCs

**Comment by Zonespace27** [comment]
> > Since it's a nonmodular TGUI change, please take it upstream first.
> 
> Tg doesn't appear to have any non adminbussed in synth race. So this would be quite useless to them

Less modular code for us to maintain, and it's not a *net loss* to TG

**Comment by Deek-Za** [comment]
> > > Since it's a nonmodular TGUI change, please take it upstream first.
> > 
> > 
> > Tg doesn't appear to have any non adminbussed in synth race. So this would be quite useless to them
> 
> Less modular code for us to maintain, and it's not a _net loss_ to TG

Unfortunately it wouldn't work upstream as it takes advantage of Skyrat code written specifically for our multitude of synthetic races. We'd have to modify TGs code further to implement it for them without it breaking(And they'd only gain the functionality for extremely niche situations). And even then when it came back downstream to us, we'd have to patch it again for it to work with our setup as ours is very different. 
I'd say redoing and adding a more additional tweaks to their code for them to reap no benefit wouldn't be in their best interests. And we'd have to maintain it on our end anyway.

**Comment by Zonespace27** [comment]
Fair 'nuff

**Comment by YakumoChen** [comment]
Would recommend an icon different than the cog, which is a universal symbol for settings or options. Is there no robot head icon? 🤖

**Comment by Deek-Za** [comment]
> Would recommend an icon different than the cog, which is a universal symbol for settings or options. Is there no robot head icon? 🤖

Fair. The reason why I went with the cog is well, cause it's associated with mechanical things and machines. And I figure worst case scenario someone one-off's "What's the cog for?" and then we explain it and move on.

 As an Alternative I can adjust it's color so it doesn't look as much like the UI element cog. Or I can do a wrench if the aesthetic would be preferred to the server staff. Here's the wrench and a rustish color applied as a combo for an example.

image

**Comment by GoldenAlpharex** [comment]
This sounds cool, you should've told me earlier, I just forgot about this.

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 1]
```diff
@@ -130,6 +131,15 @@ const CrewTableEntry = (props, context) => {
...
       </Table.Cell>
+        {
+        //SKYRAT EDIT BEGIN: Displaying robotic species Icon
+        }
+      <Table.Cell collapsing textAlign="center">
+        {is_robot ? <Icon name="wrench" color="#B7410E" size={1} /> : ""}
+      </Table.Cell>
+        {
+          //SKYRAT EDIT END
+        }
```
Comment (GoldenAlpharex) [inline tgui/packages/tgui/interfaces/CrewConsole.js]: Here, this is how to do comments in HTML.
```suggestion
       <!-- SKYRAT EDIT START - Displaying robotic species Icon -->
      <Table.Cell collapsing textAlign="center">
        {is_robot ? <Icon name="wrench" color="#B7410E" size={1} /> : ""}
      </Table.Cell>
      <!-- SKYRAT EDIT END -->
```

File: code/game/machinery/computer/crew.dm [hunk 2]
```diff
@@ -47,6 +47,7 @@
 	. += create_table_notices(list(
 		"name",
 		"job",
+		"is_robot", //SKYRAT EDIT ADD
```
Comment (GoldenAlpharex) [inline code/game/machinery/computer/crew.dm]: For these comments, please add the full `SKYRAT EDIT ADDITION - Displaying robotic species Icon`, so we know why it was added. I'm not going to copy-paste this comment, but I'm sure you know what I mean by this.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Ping me on Discord when you've addressed these things, we can get this merged, and then potentially look to see if we can't move that upstream at some point, if we think they'd need it.

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by ShoziOffical** [comment]
lovely

**Pull request merged**